pkg/controller/performanceprofile/resources.go

```diff
@@ -0,0 +1,350 @@
+package performanceprofile
+
+import (
+	"context"
+	"reflect"
+
+	configv1 "github.com/openshift/api/config/v1"
+	tunedv1 "github.com/openshift/cluster-node-tuning-operator/pkg/apis/tuned/v1"
+	mcov1 "github.com/openshift/machine-config-operator/pkg/apis/machineconfiguration.openshift.io/v1"
+
+	apiequality "k8s.io/apimachinery/pkg/api/equality"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+func mergeMaps(src map[string]string, dst map[string]string) {
+	for k, v := range src {
+		// NOTE: it will override destination values
+		dst[k] = v
+	}
+}
+
+// TODO: we should merge all create, get and delete methods
+
+func (r *ReconcilePerformanceProfile) getMachineConfigPool(name string) (*mcov1.MachineConfigPool, error) {
+	mcp := &mcov1.MachineConfigPool{}
+	key := types.NamespacedName{
+		Name:      name,
+		Namespace: metav1.NamespaceNone,
+	}
+	if err := r.client.Get(context.TODO(), key, mcp); err != nil {
+		return nil, err
+	}
+	return mcp, nil
+}
+
+func (r *ReconcilePerformanceProfile) getMutatedMachineConfigPool(mcp *mcov1.MachineConfigPool) (*mcov1.MachineConfigPool, bool, error) {
+	existing, err := r.getMachineConfigPool(mcp.Name)
+	if errors.IsNotFound(err) {
+		return mcp, true, nil
+	}
+
+	if err != nil {
+		return nil, false, err
+	}
+
+	mutated := existing.DeepCopy()
+
+	mergeMaps(mcp.Annotations, mutated.Annotations)
+	mergeMaps(mcp.Labels, mutated.Labels)
+
+	mutated.Spec = mcp.Spec
+
+	// do not update the pause and configuration fields
+	mutated.Spec.Paused = existing.Spec.Paused
```

Comment:
I think we should remove this now. it shouldn't matter anymore. Reconciling based on whether the mcp is paused or not shouldn't impact us now with the new logic.

Reply:
I am unsure it correct. for example, our code will pause the machine config pool, but the MCP that we will generate will have spec.paused false, so it will recognize the change and will unpause.

Reply:
that's a bug. the mcp we generate needs to reflect whether or not the pause is expected or not. 

we can't prevent the reconcile loop from popping early, this logic sounds like we're depending on the reconcile loop not executing too early which would result in a early unpause.